### PR TITLE
[SPARK-20127][CORE] few warning have been fixed which Intellij IDEA reported Intellij IDEA

### PR DESCRIPTION
**Pull request by dbolshak**
## What changes were proposed in this pull request?
Few changes related to Intellij IDEA inspection.

## How was this patch tested?
Changes were tested by existing unit tests

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/ui/SparkUI.scala [hunk 1]
```diff
@@ -56,7 +56,7 @@ private[spark] class SparkUI private (
   with Logging
   with UIRoot {
 
-  val killEnabled = sc.map(_.conf.getBoolean("spark.ui.killEnabled", true)).getOrElse(false)
+  val killEnabled = sc.exists(_.conf.getBoolean("spark.ui.killEnabled", true))
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/ui/SparkUI.scala]: Bringing forward some of my comments: I would probably not change this

File: core/src/main/scala/org/apache/spark/ui/UIUtils.scala [hunk 2]
```diff
@@ -513,8 +513,8 @@ private[spark] object UIUtils extends Logging {
     param
   }
 
-  def getTimeZoneOffset() : Int =
-    TimeZone.getDefault().getOffset(System.currentTimeMillis()) / 1000 / 60
+  def getTimeZoneOffset(): Int =
+    TimeZone.getDefault.getOffset(System.currentTimeMillis()) / 1000 / 60
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/ui/UIUtils.scala]: I also wouldn't remove the parens here (though the convention was always to use parens when calling no-arg Java methods?) or elsewhere. That really isn't done consistently at all anyway.

File: core/src/main/scala/org/apache/spark/ui/jobs/JobsTab.scala [hunk 3]
```diff
@@ -31,7 +31,7 @@ private[ui] class JobsTab(parent: SparkUI) extends SparkUITab(parent, "jobs") {
   val operationGraphListener = parent.operationGraphListener
 
   def isFairScheduler: Boolean =
-    jobProgresslistener.schedulingMode == Some(SchedulingMode.FAIR)
+    jobProgresslistener.schedulingMode.contains(SchedulingMode.FAIR)
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/ui/jobs/JobsTab.scala]: Likewise not sure about this change

File: core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala [hunk 4]
```diff
@@ -290,7 +290,7 @@ private[ui] class StagePage(parent: StagesTab) extends WebUIPage("stage") {
         val _taskTable = new TaskPagedTable(
           parent.conf,
           UIUtils.prependBaseUri(parent.basePath) +
-            s"/stages/stage?id=${stageId}&attempt=${stageAttemptId}",
+            s"/stages/stage?id=$stageId&attempt=$stageAttemptId",
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala]: Agree they can be omitted but not sure it helps

**Review by srowen — CHANGES_REQUESTED** [review]
I brought forward some comments but everything else looks OK. I'd be OK accepting the other changes.

File: core/src/main/scala/org/apache/spark/ui/UIUtils.scala [hunk 5]
```diff
@@ -317,7 +317,7 @@ private[spark] object UIUtils extends Logging {
     def getHeaderContent(header: String): Seq[Node] = {
       if (newlinesInHeader) {
         <ul class="unstyled">
-          { header.split("\n").map { case t => <li> {t} </li> } }
+          { header.split("\n").map(t => <li> {t} </li>)}
```
Comment (jerryshao) [inline core/src/main/scala/org/apache/spark/ui/UIUtils.scala]: I think we normally use `{}` in this code, like `map { t => xxx }`.

Comment (srowen) [inline core/src/main/scala/org/apache/spark/ui/UIUtils.scala]: That also works and gets rid of `case`. I personally prefer not opening a scope if it's not needed, and it isn't here, but it's minor as it's just a question of () vs {} and would rarely have any other impact.

**Comment by jerryshao** [comment]
There're many changes related to UI part, actually we don't have many unit tests covered in this part, so I'm afraid these change may potentially introduce regression.

File: core/src/main/scala/org/apache/spark/ui/UIUtils.scala (hunk 5 above)
Comment (dbolshak) [inline core/src/main/scala/org/apache/spark/ui/UIUtils.scala]: I removed almost changes related to UI.

File: core/src/main/scala/org/apache/spark/ui/jobs/StagesTab.scala [hunk 6]
```diff
@@ -35,7 +35,7 @@ private[ui] class StagesTab(parent: SparkUI) extends SparkUITab(parent, "stages"
   attachPage(new StagePage(this))
   attachPage(new PoolPage(this))
 
-  def isFairScheduler: Boolean = progressListener.schedulingMode == Some(SchedulingMode.FAIR)
+  def isFairScheduler: Boolean = progressListener.schedulingMode.contains(SchedulingMode.FAIR)
```
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/ui/jobs/StagesTab.scala]: Up to my knowledge, `contains` on `Option` breaks Scala 2.10 build (see https://github.com/apache/spark/pull/15393). I guess we are going to drop the support but not yet.

File: core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala [hunk 7]
```diff
@@ -786,8 +786,8 @@ private[ui] object StagePage {
       info: TaskInfo, metrics: TaskMetricsUIData, currentTime: Long): Long = {
     if (info.finished) {
       val totalExecutionTime = info.finishTime - info.launchTime
-      val executorOverhead = (metrics.executorDeserializeTime +
-        metrics.resultSerializationTime)
+      val executorOverhead = metrics.executorDeserializeTime +
+        metrics.resultSerializationTime
```
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala]: I think we can make this inlined if more commits should be pushed.

**Comment by HyukjinKwon** [comment]
Probably, I guess this should be fine. Just in my experience, IntelliJ's inspection was quite okay except the case of breaking Scala 2.10. It might be better if they can be manually tested via UI though.

**Review by dbolshak — COMMENTED** [review]
Folks, could you please go back to review? Looks like I reverted all my changes related to your comments.

**Comment by HyukjinKwon** [comment]
They are suggestions in my point of view. it doesn't necessarily mean you should follow if there are some reasons.

**Comment by dbolshak** [comment]
@HyukjinKwon could you please explain how to request to merge the PR?

File: core/src/main/scala/org/apache/spark/ui/jobs/AllStagesPage.scala [hunk 8]
```diff
@@ -34,9 +34,9 @@ private[ui] class AllStagesPage(parent: StagesTab) extends WebUIPage("") {
     listener.synchronized {
       val activeStages = listener.activeStages.values.toSeq
       val pendingStages = listener.pendingStages.values.toSeq
-      val completedStages = listener.completedStages.reverse.toSeq
+      val completedStages = listener.completedStages.reverse
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/ui/jobs/AllStagesPage.scala]: I'm neutral on this because Seq.toSeq is a no-op anyway and, theoretically, the type of completedStages could change and then this isn't a Seq, but, in practice, wont' matter.

File: core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala [hunk 9]
```diff
@@ -103,7 +103,7 @@ private[ui] class StagePage(parent: StagesTab) extends WebUIPage("stage") {
       val taskSortColumn = Option(parameterTaskSortColumn).map { sortColumn =>
         UIUtils.decodeURLParameter(sortColumn)
       }.getOrElse("Index")
-      val taskSortDesc = Option(parameterTaskSortDesc).map(_.toBoolean).getOrElse(false)
+      val taskSortDesc = Option(parameterTaskSortDesc).exists(_.toBoolean)
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala]: Still not sure about whether this is clearer. I'm neutral. @HyukjinKwon is right about `.contains` I think so I'd double check this existed in 2.10?

File: core/src/main/scala/org/apache/spark/ui/storage/RDDPage.scala [hunk 10]
```diff
@@ -69,7 +69,7 @@ private[ui] class RDDPage(parent: StorageTab) extends WebUIPage("rdd") {
     }
     val blockTableHTML = try {
       val _blockTable = new BlockPagedTable(
-        UIUtils.prependBaseUri(parent.basePath) + s"/storage/rdd/?id=${rddId}",
+        UIUtils.prependBaseUri(parent.basePath) + s"/storage/rdd/?id=$rddId",
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/ui/storage/RDDPage.scala]: I still don't this it's worth unbracing. The braces aren't hard to read and maybe prevent some misinterpretation one day after a refactoring.

File: core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala (hunk 9 above)
Comment (dbolshak) [inline core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala]: There is no `Option.contains` call any more, it was in another place.

File: core/src/main/scala/org/apache/spark/ui/storage/RDDPage.scala (hunk 10 above)
Comment (dbolshak) [inline core/src/main/scala/org/apache/spark/ui/storage/RDDPage.scala]: This one I've just reverted back.

File: core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala (hunk 9 above)
Comment (srowen) [inline core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala]: yes, I meant the question is whether .exists is in 2.10, but I am back at a desktop now and found it is, so no problem. That's not the issue, just, whether it's clearer

File: core/src/main/scala/org/apache/spark/ui/jobs/StagesTab.scala (hunk 6 above)
Comment (dbolshak) [inline core/src/main/scala/org/apache/spark/ui/jobs/StagesTab.scala]: Thanks for that comment. I did not know that.

File: core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala (hunk 9 above)
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala]: If my opinion can be helpful, personally I prefer the previous one to explicitly show the default. Actually, I intendedly use this pattern in several places.

Comment (dbolshak) [inline core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala]: Ok, I've reverted this back.

File: core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala (hunk 4 above)
Comment (dbolshak) [inline core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala]: the change was reverted.

File: core/src/main/scala/org/apache/spark/ui/jobs/StagesTab.scala (hunk 6 above)
Comment (dbolshak) [inline core/src/main/scala/org/apache/spark/ui/jobs/StagesTab.scala]: BTW, this was reverted.

File: core/src/main/scala/org/apache/spark/ui/jobs/AllStagesPage.scala (hunk 8 above)
Comment (dbolshak) [inline core/src/main/scala/org/apache/spark/ui/jobs/AllStagesPage.scala]: If it's really matter, it's better to add type to `val completedStages`, it describes your point much better then trying call no-op method. But it won't be consistent, so I would prefer to leave it as is.

**Comment by dbolshak** [comment]
Looks like all comments have been addressed.

**Comment by dbolshak** [comment]
@srowen, @HyukjinKwon could you please merge the PR if it's ok of course?

File: core/src/main/scala/org/apache/spark/ui/jobs/StagesTab.scala (hunk 6 above)
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/ui/jobs/StagesTab.scala]: It seems not reverted yet..

File: core/src/main/scala/org/apache/spark/ui/jobs/UIData.scala [hunk 11]
```diff
@@ -180,8 +180,8 @@ private[spark] object UIData {
         speculative = taskInfo.speculative
       )
       newTaskInfo.gettingResultTime = taskInfo.gettingResultTime
-      newTaskInfo.setAccumulables(taskInfo.accumulables.filter {
-        accum => !accum.internal && accum.metadata != Some(AccumulatorContext.SQL_ACCUM_IDENTIFIER)
+      newTaskInfo.setAccumulables(taskInfo.accumulables.filter { acc =>
+        !acc.internal && !acc.metadata.contains(AccumulatorContext.SQL_ACCUM_IDENTIFIER)
```
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/ui/jobs/UIData.scala]: We should revert this one too for the same reason in https://github.com/apache/spark/pull/17458/files#r108432518

File: core/src/main/scala/org/apache/spark/ui/storage/RDDPage.scala [hunk 12]
```diff
@@ -42,7 +42,7 @@ private[ui] class RDDPage(parent: StorageTab) extends WebUIPage("rdd") {
 
     val blockPage = Option(parameterBlockPage).map(_.toInt).getOrElse(1)
     val blockSortColumn = Option(parameterBlockSortColumn).getOrElse("Block Name")
-    val blockSortDesc = Option(parameterBlockSortDesc).map(_.toBoolean).getOrElse(false)
+    val blockSortDesc = Option(parameterBlockSortDesc).exists(_.toBoolean)
```
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/ui/storage/RDDPage.scala]: This also looks the same instance. Let's revert this change too.

File: core/src/main/scala/org/apache/spark/ui/WebUI.scala [hunk 13]
```diff
@@ -116,15 +116,15 @@ private[spark] abstract class WebUI(
    * @param path Path in UI to unmount.
    */
   def removeStaticHandler(path: String): Unit = {
-    handlers.find(_.getContextPath() == path).foreach(detachHandler)
+    handlers.find(_.getContextPath == path).foreach(detachHandler)
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/ui/WebUI.scala]: @dbolshak this still isn't addressing comments that have been made several times. You can appreciate why I don't think it's worth the time to make changes like this given how much discussion it takes. Here's another example. Please read _all_ previous comments and address all of them or close this.

File: core/src/main/scala/org/apache/spark/ui/jobs/AllStagesPage.scala (hunk 8 above)
Comment (dbolshak) [inline core/src/main/scala/org/apache/spark/ui/jobs/AllStagesPage.scala]: For me not clear, do you request this change back or we can leave my changes.

**Comment by dbolshak** [comment]
Corrected.
Please take a look.
There is only one change (in 2 places) related to calling toSeq which has comment, but it's not clear can I leave my change or not.

**Review by srowen — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
It looks good to me too.

**Comment by srowen** [comment]
Merged to master
